src/api/resources/collect_record.py

```diff
@@ -42,7 +42,10 @@
 class CollectRecordOwner(permissions.BasePermission):
     def has_permission(self, request, view):
         record_ids = request.data.get("ids") or []
-        if not record_ids:
+        pk = view.kwargs.get("pk")
```

Comment:
This makes me think maybe we should do a quick look at all other permissions classes to catch the same thing?

Reply:
@saanobhaai I checked sites, management and project profile I think we are OK.  Collect Record is bit of an exception because the records belong to the project and a person where the other records belong to the Project.

I also tested editing a ProjectProfile with the read-only and collector roles and was blocked